search-app/src/cmr/search/results_handlers/results_handler_util.clj

```diff
@@ -0,0 +1,59 @@
+(ns cmr.search.results-handlers.results-handler-util
+  "Provides useful functions for several of the result handlers."
+  (:require
+   [clojure.set :as s]
+   [cmr.common.util :as util]))
+
+(defn build-each-concept-association-list
+  "This function builds a map with the passed in conept-key as the key and
+  a list of concept ids from the passed in associations."
+  [concept-key associations for-concept-type]
+  (if (= :collection for-concept-type)
+    {concept-key (or (seq (util/remove-nils-empty-maps-seqs
+                           (mapv :variable-concept-id associations)))
+                     (seq (util/remove-nils-empty-maps-seqs
+                           (mapv :service-concept-id associations)))
+                     (seq (util/remove-nils-empty-maps-seqs
+                           (mapv :tool-concept-id associations)))
+                     associations)}
```

Comment:
Can there be a case where there could be multiple true values for associations? This will only ever get the first of one type of association in the case of variable, service, or tool, or be an entire list if any other associations exist.

Reply:
In this function there will ever only be 1 concept type of associations. In another function I group them by concept types and then for each type I call this function.